uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -110,27 +111,72 @@ define([
     });
 
     describe('update()', function () {
-      var attr,
-        sendPOSTMock,
-        updatePromise;
+      var requestData, promise;
+
+      beforeEach(function () {
+        requestData = mockData.createRandomLeaveRequest();
+        spyOn(LeaveRequestAPI, 'update').and.callThrough();
+        promise = LeaveRequestAPI.update(requestData);
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        promise.then(function () {
+          expect(LeaveRequestAPI.update).toHaveBeenCalled();
+        });
+      });
+    });
+
+    describe('create()', function () {
+      var requestData, promise;
+
+      beforeEach(function () {
+        requestData = mockData.createRandomLeaveRequest();
+        spyOn(LeaveRequestAPI, 'create').and.callThrough();
+        promise = LeaveRequestAPI.create(requestData);
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('calls equivalent API method', function () {
+        promise.then(function () {
+          expect(LeaveRequestAPI.create).toHaveBeenCalled();
+        });
+      });
+
+      it('id is appended to instance', function () {
+        expect(requestData.id).not.toBeDefined();
+        promise.then(function (result) {
+          expect(result.id).toBeDefined();
+        });
+      });
+    });
+
+    describe('isValid()', function () {
```

Comment:
At no point, neither in the test for `create()`, for `update()`, nor for `isValid()`, you are actually testing the instance, all you are doing is testing the API. Hence these tests are basically pointless if we keep them like this